File: app/src/main/java/me/sunzheng/mana/utils/HostUtil.java
package me.sunzheng.mana.utils;

import androidx.core.util.PatternsCompat;

public class HostUtil {
    public static String makeUp(String host, String url) {
        if (PatternsCompat.WEB_URL.matcher(url).matches())
            return url;
        else {
            return host + url;
        }
    }
}

File: app/src/main/java/me/sunzheng/mana/utils/HostUtil.kt
package me.sunzheng.mana.utils

import android.net.Uri
import android.text.TextUtils
import androidx.core.net.toUri

object HostUtil {
    @JvmStatic
    fun makeUp(host: String? = null, url: String? = null): String {
        var _host = if (host!!.endsWith("/"))
            host.substring(0, host.length - 1)
        else host
        var uri = Uri.parse(url)
        return if (!TextUtils.isEmpty(uri.host)) url!! else {
//            host + url
            (_host + url).toUri().toString()
        }
    }
}